test/macros.jl

```diff
@@ -270,6 +270,17 @@ end
             "DenseAxisArray, or SparseAxisArray.")
         @test_throws exception @variable(model, x[1:3], container=Oops)
     end
+
+    @testset "Adjoints" begin
+        model = Model()
+        @variable(model, x[1:2])
+        obj = @objective(model, Min, x' * x)
```

Comment:
For the bug to be triggered, you need something at least as complex as `x' * ones(2, 2) * x`.

Reply:
To confirm, on JuMP master:
```julia
julia> using JuMP; model = Model(); @variable(model, x[1:2]);

julia> @objective(model, Min, x' * x)
x[1]² + x[2]²

julia> typeof(@objective(model, Min, x' * x))
JuMP.GenericQuadExpr{Float64,VariableRef}
```

Reply:
I've modified the test to include a `Q` matrix and added an explicit call to `destructive_add!` in the tests.